src/Blef.GameLogic/PokerHands/TwoPairs.cs

```diff
@@ -20,8 +21,30 @@ public override bool IsOnTable(Table table)
             return firstRankCount >= 2 && secondRankCount >= 2;
         }
 
-        protected override int Value => 100 + GetRankValue(first) + GetRankValue(second);
+        protected override int GenericPokerHand => 3;
 
+        protected override int CompareWithinSameGenericPokerHand(PokerHand otherPokerHand)
+        {
+            var otherTwoPair = otherPokerHand as TwoPairs;
+
+            if (otherTwoPair == null)
+            {
+                throw new InvalidOperationException($"This method can compare only {nameof(PokerHand)} with the same GenericPokerHand value");
+            }
+
+            return CompareWithinSameGenericPokerHand(otherTwoPair);
+        }
 
+        int CompareWithinSameGenericPokerHand(TwoPairs otherTwoPair)
```

Comment:
We can change it of course to HierarchySomething, I have even started from it writing this code, but then I realised it is better cause ppl will use it in algorithms and it is even better for them to have int.
Of course now problem for me to change it to PokerHandHierarchyResult or something. I just wanted to show the idea.